app/controllers/category_experts_controller.rb

```diff
@@ -7,11 +7,9 @@ class CategoryExpertsController < ApplicationController
     :retroactive_approval?,
   ]
   before_action :ensure_needs_approval_enabled, only: [:approve_post, :unapprove_post]
+  before_action :authenticate_and_find_user, only: [:endorse, :endorsable_categories]
```

Comment:
Could you use `requires_login only: ...` here and `current_user` instead of `@user` below?

Reply:
I could do the first part, but the `@user` is not the current user, its the user to be endorsed.